frontend/src/utils/workspaceValidation.ts

```diff
@@ -0,0 +1,28 @@
+import { UseControllerProps } from 'react-hook-form'
+
+const MAX_TITLE_LENGTH = 200
+const MIN_TITLE_LENGTH = 4
+
+export const WORKSPACE_TITLE_VALIDATION_RULES: UseControllerProps['rules'] = {
+  required: 'Workspace title is required',
+  minLength: {
+    value: MIN_TITLE_LENGTH,
+    message: `Workspace title must be at least ${MIN_TITLE_LENGTH} characters`,
+  },
+  maxLength: {
+    value: MAX_TITLE_LENGTH,
+    message: `Workspace title must be at most ${MAX_TITLE_LENGTH} characters`,
+  },
+  pattern: {
+    value: /^[a-zA-Z0-9_\-./() &`;'"]*$/,
+    message: 'Workspace title cannot contain special characters',
+  },
```

Comment:
comment: i wonder if this is an artificial limitation or?

Reply:
What do you mean by artificial limitation? It's only FE limitation for now, when the BE is implemented we will need to have this validation as well!